src/main/java/com/ericsson/gerrit/plugins/eiffel/messaging/EiffelEventSender.java

```diff
@@ -89,12 +106,22 @@ public EiffelEventSender setEiffelEventType(final String eiffelType) {
         return this;
     }
 
-    private void generateAndPublish()
+    private String generateAndPublish()
             throws URISyntaxException, IOException, MissingConfigurationException,
             HttpRequestFailedException {
         assembleRequest();
         final ResponseEntity response = httpRequest.performRequest();
         verifyResponse(response);
+
+        String generatedEventId = getGeneratedEventId(response);
+        return generatedEventId;
+    }
+
+    private String getGeneratedEventId(final ResponseEntity response) {
+        JsonObject responseBody = new JsonParser().parse(response.getBody()).getAsJsonObject();
+        JsonArray events = responseBody.get("events").getAsJsonArray();
+        String generatedEventId = events.get(0).getAsJsonObject().get("id").getAsString();
```

Comment:
Are you certain that you allways have a "0"? This could cause a null pointer exception